api/resource/factory.go

```diff
@@ -42,7 +42,11 @@ func (rf *Factory) Hasher() ifc.KustHasher {
 
 // FromMap returns a new instance of Resource.
 func (rf *Factory) FromMap(m map[string]interface{}) *Resource {
-	return rf.FromMapAndOption(m, nil)
+	res, err := rf.FromMapAndOption(m, nil)
+	if err != nil {
+		log.Fatalf("failed to create resource from map: %v", err)
```

Comment:
I think the idea of the TODO was to return the error all the way to the top, e.g. completely avoid the use of `log.Fatalf` for this. That would mean `FromMap` and `FromMapWithNamespaceAndName` should the error, instead of doing `log.Fatalf` here, and likewise all the way up.

Looks like you and @koba1t already had a discussion about potentially doing that in a follow-up PR. I think that's fine to do in a followup, but in that case could we keep the TODO here? 

```	
if err != nil {
		// TODO: return err instead of log.
		log.Fatalf("failed to create resource from map: %v", err)
	}
```

Reply:
@natasha41575 Thank you for the review. Yes, I will put the TODO comment again